plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -282,11 +282,21 @@ bool GSSettingsDlg::OnCommand(HWND hWnd, UINT id, UINT code)
 			theApp.SetConfig("fba", (int)IsDlgButtonChecked(m_hWnd, IDC_FBA));
 			theApp.SetConfig("aa1", (int)IsDlgButtonChecked(m_hWnd, IDC_AA1));
 			theApp.SetConfig("mipmap", (int)IsDlgButtonChecked(m_hWnd, IDC_MIPMAP));
-			theApp.SetConfig("resx", (int)SendMessage(GetDlgItem(m_hWnd, IDC_RESX), UDM_GETPOS, 0, 0));
-			theApp.SetConfig("resy", (int)SendMessage(GetDlgItem(m_hWnd, IDC_RESY), UDM_GETPOS, 0, 0));
 			theApp.SetConfig("extrathreads", (int)SendMessage(GetDlgItem(m_hWnd, IDC_SWTHREADS), UDM_GETPOS, 0, 0));
 			theApp.SetConfig("accurate_date", (int)IsDlgButtonChecked(m_hWnd, IDC_ACCURATE_DATE));
 			theApp.SetConfig("UserHacks", (int)IsDlgButtonChecked(m_hWnd, IDC_HACKS_ENABLED));
+
+			//Clamp custom resolution values to their input range.
+			LPARAM limit_x = SendMessage(GetDlgItem(m_hWnd, IDC_RESX), UDM_GETPOS, 0, 0);
+			LPARAM limit_y = SendMessage(GetDlgItem(m_hWnd, IDC_RESY), UDM_GETPOS, 0, 0);
+			if (HIWORD(limit_x))
+				theApp.SetConfig("resx", LOWORD(limit_x) & 0xFFFF);
```

Comment:
LOWORD is 16 bits, so `& 0xFFFF` is unnecessary.

Note: While UDM_GETPOS restricts the LOWORD values to the specified range, it isn't exactly documented - it states the HIWORD is nonzero when there's an error, but doesn't state what the LOWORD is set to when there's an error. So it may be better to do the max/min clamping (but it's up to you).

Note 2: It'd probably be a good idea to restrict the software threads range as well.